Python/ast.c

```diff
@@ -14,9 +14,9 @@
 
 #define MAXLEVEL 200    /* Max parentheses level */
 
-static int validate_stmts(asdl_seq *);
-static int validate_exprs(asdl_seq *, expr_context_ty, int);
-static int validate_nonempty_seq(asdl_seq *, const char *, const char *);
+static int validate_stmts(asdl_stmt_seq *);
+static int validate_exprs(asdl_expr_seq*, expr_context_ty, int);
+static int _validate_nonempty_seq(asdl_seq *, const char *, const char *);
```

Comment:
Why? This is used as `validate_nonempty_seq` us just that the `validate_nonempty_seq` casts stuff to `asdl_seq` to avoid multiple modifications in the call site. I can do those modifications if we don't want the macro, though.

Reply:
Yeah, but since you define `validate_nonempty_seq` right after the `_validate_nonempty_seq`, we dont have to pre-declare it, right?